ts/global.d.ts

```diff
@@ -148,29 +217,220 @@ declare const enum Pitch {
 }
 
 declare class MidiEvent {
+    /** Sends the event. */
     send(): void;
+
+    /**
+     * Sends the event after a certain number of milliseconds has elapsed.
+     *
+     * @param ms - The number of milliseconds to wait before sending the event.
+     */
+    sendAfterMilliseconds(ms: number): void;
+
+    /**
+     * Sends the event after a certain number of beats.
+     *
+     * @param beat - The number of beats to wait before sending the event.
+     */
+    sendAtBeat(beat: number): void;
+
+    /** Prints the event to the plug-in console. */
+    trace(): void;
+
+    /** @returns A string representation of the event. */
+    toString(): string;
+
+    /** The MIDI channel of the event. Can be a value between 1 and 16. */
+    channel: number;
```

Comment:
It might be nice to have a const enum for MIDI channel as well.

Reply:
Hmm, I feel like that could be a bit overkill in this case since the channel is already a number intrinsically. (That, and if anyone sets/uses the MIDI channel, it's probably more likely to be coming from a user parameter rather than hard coded in the script, making the type checking less useful.)

Reply:
That's a fair point - I guess it's still possible to use the CC enum to refer to numbers, and that's where you'd want to use the enums.

Reply:
Yeah, if I were to do a type restriction on channel, I'd probably go with a union type over an enum e.g. "type MidiChannel = 1 | 2 | 3 | 4" and so on

Reply:
The reason I went with an enum is that some of the channels are named so you can do things like `CC.Hold` or `CC.Modulation` (e.g. https://github.com/sdcooke/Scripter/blob/master/ts/pad.ts#L83)

Reply:
Oh, hmm, I didn't know about channel 10 being reserved for percussion in General MIDI. (My knowledge of the MIDI spec is not terribly deep haha.) Totally agree that using an enum for CC's makes a lot of sense, though. (Was going to fill out the CC enum a bit more; I'll try to get the commit in this evening after work.)

Reply:
Oops I've just realised I was getting mixed up and thought we were talking about CCs. Sorry about that - you're right, the channel makes sense as a number - the only reason to do an enum would be to limit the range. Channel 10 being drums takes me back to messing around with general midi a long time ago! Sorry if I wasted any of your time.

Reply:
No worries!